presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/BridgingHiveMetastore.java

```diff
@@ -267,14 +267,9 @@ private void alterTable(String databaseName, String tableName, org.apache.hadoop
     }
 
     @Override
-    public void addPartitions(String databaseName, String tableName, List<Partition> partitions)
+    public void addPartitions(String databaseName, String tableName, List<PartitionWithStatistics> partitions)
     {
-        delegate.addPartitions(
-                databaseName,
-                tableName,
-                partitions.stream()
-                        .map(ThriftMetastoreUtil::toMetastoreApiPartition)
-                        .collect(Collectors.toList()));
+        delegate.addPartitions(databaseName, tableName, partitions);
```

Comment:
It is not possible, given that our internal metastore has it's own representation of Partition and Statistics. Currently we override the `HiveMetastore` interface. In order to do this conversion in `BridgingHiveMetastore` we will need to override `ExtendedHiveMetastore` in our library. Previously we had this discussion with @electrum , and we decided to go this way to avoid making a mess.